### PR TITLE
release 4.77.0 changelog

### PR DESCRIPTION
**Pull request by c2thorn**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,43 @@
 ## 4.77.0 (Unreleased)
 
+NOTES:
+* vpcaccess: reverted the ability to update the number of instances for resource `google_vpc_access_connector` ([#15313](https://github.com/hashicorp/terraform-provider-google/pull/15313))
```
Comment (slevenick) [inline CHANGELOG.md:4]: Do we want a description here of the error users would have seen so they know what is fixed?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,43 @@
...
+* compute: added `local_ssd_recovery_timeout` field to `google_compute_instance` resource ([#15366](https://github.com/hashicorp/terraform-provider-google/pull/15366))
+* compute: added `local_ssd_recovery_timeout` field to `google_compute_instance_template` resource ([#15366](https://github.com/hashicorp/terraform-provider-google/pull/15366))
+* compute: added `network_interface.ipv6_access_config.external_ipv6_prefix_length` to `google_compute_instance` ([#15358](https://github.com/hashicorp/terraform-provider-google/pull/15358))
+* compute: added `network_interface.ipv6_access_config.name` to `google_compute_instance` ([#15358](https://github.com/hashicorp/terraform-provider-google/pull/15358))
+* compute: added a new type `GLOBAL_MANAGED_PROXY` for the filed `purpose` in the resource `google_compute_subnetwork` ([#15345](https://github.com/hashicorp/terraform-provider-google/pull/15345))
+* compute: added field `instance_lifecycle_policy` to `google_compute_instance_group_manager` and `google_compute_region_instance_group_manager` (ga) ([#15322](https://github.com/hashicorp/terraform-provider-google/pull/15322))
```
Comment (slevenick) [inline CHANGELOG.md:20]:
```suggestion
* compute: added field `instance_lifecycle_policy` to `google_compute_instance_group_manager` and `google_compute_region_instance_group_manager` ([#15322](https://github.com/hashicorp/terraform-provider-google/pull/15322))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,43 @@
...
+
+FEATURES:
+* **New Resource:** `google_document_ai_warehouse_document_schema` ([#15326](https://github.com/hashicorp/terraform-provider-google/pull/15326))
+* **New Resource:** `google_document_ai_warehouse_location` ([#15326](https://github.com/hashicorp/terraform-provider-google/pull/15326))
+
+IMPROVEMENTS:
+* alloydb: added `continuous_backup_config` and `continuous_backup_info` fields to `cluster` resource ([#15370](https://github.com/hashicorp/terraform-provider-google/pull/15370))
+* cloudbuild: removed the validation function for the values of `machine_type` field on the `google_cloudbuild_trigger` resource ([#15357](https://github.com/hashicorp/terraform-provider-google/pull/15357))
+* compute: add future_limit in quota exceeded error details for compute resources. ([#15346](https://github.com/hashicorp/terraform-provider-google/pull/15346))
+* compute: added `ipv6EndpointType` and `ipVersion` to `google_compute_address` ([#15358](https://github.com/hashicorp/terraform-provider-google/pull/15358))
```
Comment (slevenick) [inline CHANGELOG.md:14]:
```suggestion
* compute: added `ipv6_endpoint_type` and `ip_version` to `google_compute_address` ([#15358](https://github.com/hashicorp/terraform-provider-google/pull/15358))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,43 @@
...
+* compute: added `local_ssd_recovery_timeout` field to `google_compute_instance` resource ([#15366](https://github.com/hashicorp/terraform-provider-google/pull/15366))
+* compute: added `local_ssd_recovery_timeout` field to `google_compute_instance_template` resource ([#15366](https://github.com/hashicorp/terraform-provider-google/pull/15366))
+* compute: added `network_interface.ipv6_access_config.external_ipv6_prefix_length` to `google_compute_instance` ([#15358](https://github.com/hashicorp/terraform-provider-google/pull/15358))
+* compute: added `network_interface.ipv6_access_config.name` to `google_compute_instance` ([#15358](https://github.com/hashicorp/terraform-provider-google/pull/15358))
+* compute: added a new type `GLOBAL_MANAGED_PROXY` for the filed `purpose` in the resource `google_compute_subnetwork` ([#15345](https://github.com/hashicorp/terraform-provider-google/pull/15345))
```
Comment (slevenick) [inline CHANGELOG.md:19]:
```suggestion
* compute: added a new type `GLOBAL_MANAGED_PROXY` for the field `purpose` in the resource `google_compute_subnetwork` ([#15345](https://github.com/hashicorp/terraform-provider-google/pull/15345))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,43 @@
...
+* compute: added field `instance_lifecycle_policy` to `google_compute_instance_group_manager` and `google_compute_region_instance_group_manager` (ga) ([#15322](https://github.com/hashicorp/terraform-provider-google/pull/15322))
+* compute: added protocol type: UNSPECIFIED in `google_compute_backend_service` as per [release note](https://cloud.google.com/load-balancing/docs/release-notes#July_24_2023)
+ ([#15328](https://github.com/hashicorp/terraform-provider-google/pull/15328))
+* compute: made `network_interface.ipv6_access_config.external_ipv6` configurable in `google_compute_instance` ([#15358](https://github.com/hashicorp/terraform-provider-google/pull/15358))
```
Comment (slevenick) [inline CHANGELOG.md:23]: Should this be in the more common form of: added x field to y resource?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,43 @@
...
+* container: added `enable_k8s_beta_apis.enabled_apis` field to `google_container_cluster` ([#15320](https://github.com/hashicorp/terraform-provider-google/pull/15320))
+* container: added `node_config.host_maintenance_policy` field to `google_container_cluster` and `google_container_node_pool` ([#15347](https://github.com/hashicorp/terraform-provider-google/pull/15347))
+* container: added `placement_policy.policy_name` field to `google_container_node_pool` resource ([#15367](https://github.com/hashicorp/terraform-provider-google/pull/15367))
+* container: allowed `enabled_private_endpoint` to be settable on creation for PSC-based clusters ([#15361](https://github.com/hashicorp/terraform-provider-google/pull/15361))
+* container: supported master_global_access_config for flexible cluster ([#15369](https://github.com/hashicorp/terraform-provider-google/pull/15369))
```
Comment (slevenick) [inline CHANGELOG.md:28]: Add ` around field name, what is the resource here?

File: CHANGELOG.md (hunk 1 above)
Comment (slevenick) [reply CHANGELOG.md:4]: I guess this happens at the end as well, so nevermind

**Review by slevenick — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (c2thorn) [reply CHANGELOG.md:23]: The field existed before as output-only, so I don't think that would work here. Open to better wording though

Comment (slevenick) [reply CHANGELOG.md:23]: Oh weird, ok

File: CHANGELOG.md (hunk 6 above)
Comment (c2thorn) [reply CHANGELOG.md:28]:
```suggestion
* container: unsuppressed `private_cluster_config ` when `master_global_access_config` is set in `google_container_cluster` ([#15369](https://github.com/hashicorp/terraform-provider-google/pull/15369))
```

**Comment by aweberlopes** [comment]
Why is CMEK for CF2 not part of this release?

**Comment by c2thorn** [comment]
> Why is CMEK for CF2 not part of this release?

It did not make the cut date for last week, and will be included in next week's release. If you want to try it out sooner, you can build the version at https://github.com/hashicorp/terraform-provider-google-beta/pull/6004, with warning that it is not yet an official build.

**Review by slevenick — APPROVED** [review]

**Pull request merged**